pypeit/spectrographs/mmt_bluechannel.py

```diff
@@ -301,17 +401,43 @@ def check_frame_type(self, ftype, fitstbl, exprng=None):
         good_exp = framematch.check_frame_exptime(fitstbl['exptime'], exprng)
         if ftype == 'bias':
             return fitstbl['idname'] == 'zero'
-        if ftype in ['science', 'standard']:
-            return good_exp & (fitstbl['lampstat01'] == 'off') & (fitstbl['idname'] == 'object')
-        if ftype in ['arc', 'tilt']:
+        if ftype == 'dark':
+            return fitstbl['idname'] == 'dark'
+        if ftype in ['science']:
```

Comment:
I would change these `in` conditionals to `==` when there is only one option.  I.e., change to `if ftype == 'science':`.

Reply:
yeah, i was moving things in and out of lists as i was finding new edge cases in the archived data. i'll clean that up.

Reply:
that said, the optimal solution is to use `match`-`case` and require python >=3.10...

Reply:
Ha!  I had exactly the same thought, but decided against the suggestion :)

Reply:
it's worth discussing making the move to require that. 3.11 is consistently at least 5% faster than 3.9 for every reduction i've compared. maybe target the `1.13.0` release for that.